core/upgrader/impl/src/controllers/disaster_recovery.rs

```diff
@@ -88,7 +101,21 @@ impl DisasterRecoveryController {
         if !is_controller(&caller) {
             Err(UpgraderApiError::NotController)?
         } else {
-            self.disaster_recovery_service.set_accounts(
+            self.disaster_recovery_service
+                .set_accounts(input.accounts.into_iter().map(Into::into).collect())
+        }
+    }
+
+    fn set_disaster_recovery_accounts_and_assets(
+        &self,
+
+        input: upgrader_api::SetDisasterRecoveryAccountsAndAssetsInput,
+    ) -> ApiResult {
+        let caller = caller();
+        if !is_controller(&caller) {
+            Err(UpgraderApiError::NotController)?
+        } else {
+            self.disaster_recovery_service.set_accounts_and_assets(
```

Comment:
nit: i would drop the else and simply return early if the `!is_controller` branch is reached